src/include/distributed/distributed_planner.h

```diff
@@ -85,10 +85,31 @@ typedef struct JoinRestriction
 	RelOptInfo *outerrel;
 } JoinRestriction;
 
+typedef struct FastPathRestrictionContext
+{
+	bool fastPathRouterQuery;
+
+	/*
+	 * While calculating fastPathRouterQuery, we could sometimes be
+	 * able to extract the distribution key value as well (such as when
+	 * there are no prepared statements). Could be NULL when the distribution
+	 * key contains parameter, so check for it before using.
+	 */
+	Const *distributionKeyValue;
+}FastPathRestrictionContext;
+
 typedef struct PlannerRestrictionContext
 {
 	RelationRestrictionContext *relationRestrictionContext;
 	JoinRestrictionContext *joinRestrictionContext;
+
+	/*
+	 * When the query is qualified for fast path, we don't have
+	 * the RelationRestrictionContext and JoinRestrictionContext
+	 * since those are dependent to calling standard_planner.
+	 * Instead, we keep this struct to pass some extra information.
+	 */
+	FastPathRestrictionContext *fastPathRestrictionContext;
```

Comment:
If we're going to pass `planningContext` down to the lower functions like `CreateRouterPlan()`, than I'd suggest to move `FastPathRestrictionContext ` into `planningContext`, with the name `FastPathPlanningContext` or such.

I've two points for this:

1) `PlannerRestrictionContext ` is required because `standard_planner` is called and we get into restrictions hooks. In fast-path case, we're not calling standard_planner but still  calling `Push/Pop` planner contexts. We could improve it later, and makes it easier to read.

2) `FastPathRestrictionContext ` does not fit into `PlannerRestrictionContext ` well because they are generated completely differently and could live as different structs.